src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -24,45 +24,28 @@ using namespace Eigen;
 using namespace shogun;
 using namespace std;
 
-static vector<float64_t> make_vector(int32_t size, float64_t val)
+CLeastAngleRegression::CLeastAngleRegression(bool lasso) :
+	CLinearMachine(), m_lasso(lasso),
+	m_max_nonz(0), m_max_l1_norm(0)
 {
-	vector<float64_t> result(size);
-	fill(result.begin(), result.end(), val);
-	return result;
+	m_epsilon = CMath::MACHINE_EPSILON;
+	SG_ADD(&m_epsilon, "epsilon", "Epsilon for early stopping", MS_AVAILABLE);
+	SG_ADD(&m_max_nonz, "max_nonz", "Max number of non-zero variables", MS_AVAILABLE);
+	SG_ADD(&m_max_l1_norm, "max_l1_norm", "Max l1-norm of estimator", MS_AVAILABLE);
 }
 
-static void plane_rot(float64_t x0, float64_t x1,
-	float64_t &y0, float64_t &y1, SGMatrix<float64_t> &G)
+CLeastAngleRegression::~CLeastAngleRegression()
 {
-	G.zero();
-	if (x1 == 0)
-	{
-		G(0, 0) = G(1, 1) = 1;
-		y0 = x0;
-		y1 = x1;
-	}
-	else
-	{
-		float64_t r = CMath::sqrt(x0*x0+x1*x1);
-		float64_t sx0 = x0 / r;
-		float64_t sx1 = x1 / r;
-
-		G(0,0) = sx0;
-		G(1,0) = -sx1;
-		G(0,1) = sx1;
-		G(1,1) = sx0;
-
-		y0 = r;
-		y1 = 0;
-	}
+	
 }
 
-static void find_max_abs(const vector<float64_t> &vec, const vector<bool> &ignore_mask,
-	int32_t &imax, float64_t& vmax)
+template <typename ST>
+void CLeastAngleRegression::find_max_abs(const std::vector<ST> &vec, const std::vector<bool> &ignore_mask,
```

Comment:
is this method hidden from SWIG?
@Saurabh7 it contains a std::vector which we usually try to avoid in interfaces